pulp_smash/tests/pulp3/file/api_v3/test_crud_importers.py

```diff
@@ -0,0 +1,99 @@
+# coding=utf-8
+"""Tests that CRUD importers."""
+import unittest
+
+from requests.exceptions import HTTPError
+
+from pulp_smash import api, config, selectors
+from pulp_smash.tests.pulp3.constants import FILE_IMPORTER_PATH, REPO_PATH
+from pulp_smash.tests.pulp3.file.api_v3.utils import (
+    gen_importer,
+    modify_importer_down_policy,
+    modify_importer_sync_mode,
+)
+from pulp_smash.tests.pulp3.pulpcore.utils import gen_repo
+from pulp_smash.tests.pulp3.utils import get_auth
+
+
+class CRUDImportersTestCase(unittest.TestCase):
+    """CRUD importers."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables.
+
+        In order to create an importer a repository has to be created first.
+        """
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.client.request_kwargs['auth'] = get_auth()
+        cls.importer = {}
+        cls.repo = cls.client.post(REPO_PATH, gen_repo())
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean class-wide variable."""
+        cls.client.delete(cls.repo['_href'])
+
+    @selectors.skip_if(bool, 'repo', False)
+    def test_01_create_importer(self):
+        """Create an importer."""
+        body = gen_importer()
+        body['repository'] = self.repo['_href']
+        type(self).importer = self.client.post(FILE_IMPORTER_PATH, body)
+
+    @selectors.skip_if(bool, 'importer', False)
+    def test_02_read_importer(self):
+        """Read an importer by its href."""
+        importer = self.client.get(self.importer['_href'])
+        self.assertEqual(self.importer['name'], importer['name'])
```

Comment:
This test asserts that the attributes **returned** when the repository was created can be fetched again. It doesn't tell us that the attributes **sent** when the repository was created can be fetched again. This test is still useful as is, because it demonstrates that HTTP GET requests to importers don't blow up. But this test isn't as useful as it could be.

Note that `pulp_smash.tests.pulp3.pulpcore.api_v3.test_crud_users` is much better written than `pulp_smash.tests.pulp3.pulpcore.api_v3.test_repos`, for this reason.

Also, it would be even better if we could iterate through a bunch of attributes and verify that they're correctly set, instead of just spot-checking the name attribute.